### DISCUSSION TITLE
Markdwon to PDF using XeLatex for Multilingual Document

### DISCUSSION THREAD

**Discussion by milavdabgar** [jgm/pandoc#10166] [General]
Hello everyone,

I’ve been using markdown to create notes for my students, which I then convert to PDF using Pandoc and either PDFLaTeX or XeLaTeX. Everything worked smoothly when I was working with English-only content.

However, I started creating bilingual notes in English and Gujarati, and I’ve run into some issues.

I've managed to resolve most of the problems, but there are two remaining challenges:

1. Gujarati text within code fences isn't rendering correctly. For example:

 ```python
 print("ટાઇમ કોમ્પ્લેક્ષીટી")
 ```

2. Mermaid diagrams inside code fences aren't rendering as diagrams.

If anyone has experience with multilingual document conversion using Pandoc and can help, I'd really appreciate it!

GitHub repo: [Repo](https://github.com/milavdabgar/biling-latex/tree/main)

**Comment by jgm** [comment]
"isn't rendering properly" -- can you describe the precise problem?

"Mermaid diagrams" -- pandoc doesn't have native support for these, though there are third-party filters that may help.

**Reply by milavdabgar** [reply]
![Screenshot_20240910-220831_GitHub.jpg](https://github.com/user-attachments/assets/c0318c34-6443-43cd-aacd-ee40b49bf6f9)

![Screenshot_20240910-220854_GitHub.jpg](https://github.com/user-attachments/assets/756c78fe-2ec8-418e-b945-4851cf8af1e6)

Character not found in particular font, for code fences.

The error was Gujarati characters, not found in Moto Sans Mono font.

I've tried several way to deal with multiple languages in code blocks, but either english compiles succesfull or gujarati, but not both.

**Reply by jgm** [reply]
The solution is to use a different font for code.  You need to find a monospace font that has glyphs for both languages.  (Use `-Vmonofont="font name"` on command line.)

**Reply by milavdabgar** [reply]
Okay, Will try it and will update
